strawberryfields/configuration.py

```diff
@@ -15,138 +15,229 @@
 This module contains the :class:`Configuration` class, which is used to
 load, store, save, and modify configuration options for Strawberry Fields.
 """
-import os
 import logging as log
+import os
 
 import toml
 from appdirs import user_config_dir
 
 log.getLogger()
 
 
-DEFAULT_CONFIG = {
-    "api": {
-        "authentication_token": "",
-        "hostname": "localhost",
-        "use_ssl": True,
-        "port": 443,
-        "debug": False}
-}
+class ConfigurationError(Exception):
+    """Exception used for configuration errors"""
+
+
+def load_config(filename="config.toml", **kwargs):
+    """Load configuration from keyword arguments, configuration file or
+    environment variables.
 
-BOOLEAN_KEYS = ("debug", "use_ssl")
+    .. note::
+
+        The configuration object (a nested dictionary) will be created based
+        on the following (order defines the importance, going from most
+        important to least important):
+
+        1. keyword arguments passed to ``load_config``
+        2. data contained in environmental variables (if any)
+        3. data contained in a configuration file (if exists)
+
+    Keyword arguments:
+        filename (str): the name of the configuration file to look for
+        authentication_token (str): the token to be used for user
+            authentication
+        hostname (str): the name of the host to connect to
+        use_ssl (bool): specifies if requests should be sent using SSL
+        port (int): the port to be used when connecting to the remote service
+        debug (bool): determines if the debugging mode is requested
```

Comment:
I'm thinking it might be best to move all the configuration options to the module docstring. This has several advantages:

* Less redundancy --- configuration options only need to be specified in one place, and linked from everywhere else

* We can use Sphinx glossary notation to add more details

* It is user facing, whereas these docstrings aren't

Reply:
I've added them to the module docstring :) Added the list of environment variables, let me know if you think they need more adjustments :)